sdk/face/azure-ai-vision-face/samples/samples_face_detection.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+FILE: sample_face_detection.py
+
+DESCRIPTION:
+    This sample demonstrates how to detect faces and analyze faces from an image or binary data.
+
+USAGE:
+    python sample_face_detection.py
+
+    Set the environment variables with your own values before running this sample:
+    1) AZURE_FACE_API_ENDPOINT - the endpoint to your Face resource.
+    2) AZURE_FACE_API_ACCOUNT_KEY - your Face API key.
+"""
+import os
+
+from dotenv import find_dotenv, load_dotenv
+
+from shared.constants import (
+    CONFIGURATION_NAME_FACE_API_ACCOUNT_KEY,
+    CONFIGURATION_NAME_FACE_API_ENDPOINT,
+    DEFAULT_FACE_API_ACCOUNT_KEY,
+    DEFAULT_FACE_API_ENDPOINT,
+    TestImages,
+)
+from shared import helpers
+from shared.helpers import beautify_json, get_logger
+
+
+class DetectFaces():
+    def __init__(self):
+        load_dotenv(find_dotenv())
+        self.endpoint = os.getenv(CONFIGURATION_NAME_FACE_API_ENDPOINT, DEFAULT_FACE_API_ENDPOINT)
+        self.key = os.getenv(CONFIGURATION_NAME_FACE_API_ACCOUNT_KEY, DEFAULT_FACE_API_ACCOUNT_KEY)
+        self.logger = get_logger("sample_face_detection")
+
+    def detect(self):
+        from azure.core.credentials import AzureKeyCredential
+        from azure.ai.vision.face import FaceClient
+        from azure.ai.vision.face.models import (
+                FaceDetectionModel, FaceRecognitionModel, FaceAttributeTypeDetection03, FaceAttributeTypeRecognition04)
+
+        with FaceClient(endpoint=self.endpoint, credential=AzureKeyCredential(self.key)) as face_client:
+            sample_file_path = helpers.get_image_path(TestImages.IMAGE_DETECTION_5)
+            result = face_client.detect(  # type: ignore
+                helpers.read_file_content(sample_file_path),
+                detection_model=FaceDetectionModel.DETECTION_03,
+                recognition_model=FaceRecognitionModel.RECOGNITION_04,
+                return_face_id=True,
+                return_face_attributes=[
+                    FaceAttributeTypeDetection03.HEAD_POSE,  # type: ignore  # pyright: ignore[reportAttributeAccessIssue]  # noqa: E501
+                    FaceAttributeTypeDetection03.MASK,  # type: ignore  # pyright: ignore[reportAttributeAccessIssue]  # noqa: E501
+                    FaceAttributeTypeRecognition04.QUALITY_FOR_RECOGNITION],  # type: ignore  # pyright: ignore[reportAttributeAccessIssue]  # noqa: E501
```

Comment:
What are all the errors here? We need to fix this in the library code, we can't expect users to add ignores themselves.

Reply:
Here's the detail about these errors. It looks like Pyright only supports static enum declaration. Probably we have to copy the definition here. Not sure if it's a good idea.
* samples/samples_face_detection.py:57: error: "Type[FaceAttributeTypeDetection03]" has no attribute "HEAD_POSE"  [attr-defined]
* /mnt/vss/_work/1/s/sdk/face/azure-ai-vision-face/samples/samples_face_detection.py:57:50 - error: Cannot access member "HEAD_POSE" for type "Enum"
    Member "HEAD_POSE" is unknown (reportAttributeAccessIssue)

Reply:
Try the steps in this comment and let me if it resolves these errors as well

https://github.com/Azure/azure-sdk-for-python/pull/35384#discussion_r1595692069

Reply:
The issue is still unresolved, and I think the reason is that I **dynamically** pick up some of `FaceAttributeType`'s member as members of this new enum class, `FaceAttributeTypeDetection01`, `FaceAttributeTypeDetection03`, `FaceAttributeTypeRecognition03` and `FaceAttributeTypeRecognition04`, which is not supported by mypy and pyright.
The reason I do that is that each customized attribute Eum class is a subset of `FaceAttributeType`. I don't want to redundantly declare the same (name, value) in each Enum class. I'd like to **directly** reference the `FaceAttributeType`'s declaration.
Please let me know if you have any suggestions to resolve these errors.


Here's why we decided to declare these 4 more Enum classes ...

This API (/detect) is quite complex. We have 13 face attributes (see [the definition](https://github.com/zihyunting/azure-sdk-for-python/blob/azure-ai-vision-face-test-cases-and-samples/sdk/face/azure-ai-vision-face/azure/ai/vision/face/models/_enums.py#L46C1-L46C71)), 3 detection models and 4 recognition models. While doing face detection, the input detection and recognition models will be used, and different models support different face attributes. However, **no model supports all attributes.** Some of them are supported by detection_01, some by detection_03, recognition_03, and recogition_04.

Thus, we'd like to add these four customized Enum classes, `FaceAttributeTypeDetection01`, `FaceAttributeTypeDetection03`, `FaceAttributeTypeRecognition03`, and `FaceAttributeTypeRecognition04`, to help users use this method more easily. For example, if you choose detection_03 and recognition_04, and then the attributes you can use are all in `FaceAttributeTypeDetection03` and `FaceAttributeTypeRecogition04`.

We understand that it may be easier to use if we split the /detect method into different model combinations. However, the total number of combinations will be up to 24 (2 method * 3 detection models * 4 recognition models). For now, **we agreed it could be shipped in a preview release, but we have to have a discussion with the stewardship board before GA**. See the discussion https://github.com/Azure/azure-sdk-for-python/pull/35384#discussion_r1594820942.

Reply:
Yes, the issue is that the enum members are dynamically set:

> azure/ai/vision/face/models/_patch.py:14: error: Second argument of Enum() must be string, tuple, list or dict literal for mypy to determine Enum members  [misc]

Would you be open to doing something like the below? mypy/pyright can understand this, we define the enum correctly, and we can still use the FaceAttributeType.

```python
class FaceAttributeTypeDetection01(str, Enum, metaclass=CaseInsensitiveEnumMeta):
    """Available attribute options for detection_01 model."""
    ACCESSORIES = FaceAttributeType.ACCESSORIES.value
    BLUR = FaceAttributeType.BLUR.value
    EXPOSURE = FaceAttributeType.EXPOSURE.value
    GLASSES = FaceAttributeType.GLASSES.value
    HEAD_POSE = FaceAttributeType.HEAD_POSE.value
    NOISE = FaceAttributeType.NOISE.value
    OCCLUSION = FaceAttributeType.OCCLUSION.value
```

Reply:
Sure. I re-wrote the enum definitions based on your suggestion and removed all ignore flags from samples.